### DISCUSSION TITLE
Mailgun emails not being triggered in production

### DISCUSSION THREAD

**Discussion by srestraj** [vercel/community#6524] [Help]
### Summary

I'm reaching out to seek assistance with an issue I'm encountering in my production deployment on Vercel.

I have a Next.js 14 project where I've integrated the Mailgun API to send emails. The integration works perfectly fine when I'm running the project locally. However, after deploying to production on Vercel, I'm consistently encountering an "Unauthorized" error when attempting to send emails via the Mailgun API.

I know that this "Unauthorized" error is being thrown by the Mailgun API, but we've already added Vercel's IP, i.e. 76.76.21.21 to our domain's whitelist on Mailgun, but still no luck.

I've thoroughly checked and verified the API keys, environment variables and permissions and they're all fine. I've also included my code below for reference.

```tsx
import { NextRequest, NextResponse } from "next/server";
import FormData from "form-data";
import Mailgun from "mailgun.js";

const { MAILGUN_EMAIL, MAILGUN_TO_EMAIL, MAILGUN_API, MAILGUN_DOMAIN } =
  process.env;

export async function POST(request: NextRequest, res: NextResponse) {
  const { name, email, ticketCount, attendingAs, exhibitorDetails } =
    await request.json();

  const mailgun = new Mailgun(FormData);

  try {
    const mg = mailgun.client({
      username: "api",
      key: MAILGUN_API as string,
    });

    const mailData = {
      from: MAILGUN_EMAIL,
      to: MAILGUN_TO_EMAIL,
      subject: `${name} just signed up for the Expo`,
      text: `${name} just signed up for ${ticketCount} tickets as ${attendingAs} for the expo with email: ${email}.`,
      html: `
      <body>
        <div style="background: #f7f7f7; padding: 20px 0; margin:0">
           body content here
        </div>
      </body>`,
    };

    await new Promise((resolve, reject) => {
      mg.messages
        .create(MAILGUN_DOMAIN as string, mailData)
        .then((message) => {
          resolve(message);
        })
        .catch((error) => {
          reject(error);
          return NextResponse.json({ error: error }, { status: error.status });
        });
    });
    return NextResponse.json(
      {
        message:
          "Thank you for your interest in attending the expo. We hope to see you there.",
      },
      { status: 200 }
    );
  } catch (error: any) {
    return NextResponse.json({ error: error }, { status: error.status });
  }
}
```
Could you please assist me in troubleshooting this issue? Any insights or guidance you can provide would be greatly appreciated.
If you require any further information or access to my Vercel project for troubleshooting purposes, please let me know, and I'll be more than happy to provide it.

### Example

https://www.mewingapp.co/mewing-expo

### Steps to Reproduce

You can reproduce the error by going to this [link](https://www.mewingapp.co/mewing-expo) and try submitting the form.
![image](https://github.com/vercel/community/assets/25652144/1a5a2594-3893-4d87-b6c7-62adfa3a0c8b)

**Comment by tafsirc** [comment]
Same problem facing right now but no solution here. Did you solved yours?

**Reply by srestraj** [reply]
Hi @ctafsiras, so I came across one Vercel [article](https://vercel.com/guides/how-to-allowlist-deployment-ip-address), which suggests that on the free plan, we don't have the option to create custom IP for our domains as Vercel deployments use a range of IP addresses that can't be easily specified for Mialgun's IP Allowlist. So, we have two options:
 * Allow all internet traffic (not recommended): This is the simplest solution but poses security risks. If you must go this route, you can add "0.0.0.0/0" to your Allowlist.
 * Use Vercel Secure Compute (for Enterprise plans): This creates a private network with dedicated IP addresses for your deployment. You can then add these IPs to your Mailgun Allowlist for enhanced security.

**Reply by tafsirc** [reply]
I have added 0.0.0.0/0 in allowlist of mailgun. but still it is not working. Right now I cannot buy Enterprise plan of vercel. can you help me any other way to solve it? or some suggestions?

**Reply by srestraj** [reply]
We decided to change our hosting provider to Hostinger since our team didn't want to switch the email providers. If you can switch to a different email provider, you could try going for Zoho for your emails instead of Mailgun. It's much easier to setup and works perfectly fine with Vercel too, without any errors.
